frontend/packages/console-app/src/components/data-view/ResourceDataView.tsx

```diff
@@ -0,0 +1,208 @@
+import * as React from 'react';
+import {
+  ResponsiveAction,
+  ResponsiveActions,
+  SkeletonTableBody,
+} from '@patternfly/react-component-groups';
+import { Bullseye, Pagination, Tooltip } from '@patternfly/react-core';
+import {
+  DataView,
+  DataViewState,
+  DataViewTable,
+  DataViewTextFilter,
+  DataViewToolbar,
+} from '@patternfly/react-data-view';
+import DataViewFilters from '@patternfly/react-data-view/dist/cjs/DataViewFilters';
+import { ColumnsIcon } from '@patternfly/react-icons';
+import { InnerScrollContainer, Tbody, Td, Tr } from '@patternfly/react-table';
+import { useTranslation } from 'react-i18next';
+import {
+  ColumnLayout,
+  K8sResourceCommon,
+} from '@console/dynamic-plugin-sdk/src/extensions/console-types';
+import { createColumnManagementModal } from '@console/internal/components/modals';
+import { TableColumn } from '@console/internal/module/k8s';
+import { EmptyBox } from '@console/shared/src/components/empty-state/EmptyBox';
+import { StatusBox } from '@console/shared/src/components/status/StatusBox';
+import { DataViewLabelFilter } from './DataViewLabelFilter';
+import { ResourceFilters, GetDataViewRows } from './types';
+import { useResourceDataViewData } from './useResourceDataViewData';
+import { useResourceDataViewFilters } from './useResourceDataViewFilters';
+
+export type ResourceDataViewProps<TData, TCustomRowData, TFilters> = {
+  label?: string;
+  data: TData[];
+  loaded: boolean;
+  loadError?: any;
+  columns: TableColumn<TData>[];
+  columnLayout?: ColumnLayout;
+  columnManagementID?: string;
+  initialFilters: TFilters;
+  additionalFilterNodes?: React.ReactNode[];
+  matchesAdditionalFilters?: (resource: TData, filters: TFilters) => boolean;
+  getDataViewRows: GetDataViewRows<TData, TCustomRowData>;
+  customRowData?: TCustomRowData;
+  showNamespaceOverride?: boolean;
+  hideNameLabelFilters?: boolean;
+  hideLabelFilter?: boolean;
+  hideColumnManagement?: boolean;
+  mock?: boolean;
+};
+
+/**
+ * Console DataView component based on PatternFly DataView.
+ */
+export const ResourceDataView = <
```

Comment:
Shall we add unit test for the new component?

Reply:
@yapei I think we can skip adding unit tests for this component at this point in time.

The main goal of this PR was to refactor original DataView PoC code from #14897 and #15057 without putting this component into any use. This is the initial step towards using PatternFly DataView code in Console.

Any immediate issues found by @rhamilto via CronTab plugin testing should be covered in this PR. Once this PR merges, the next step should be putting this component into use on select Console data tables and resolve any other issues we find along the way.

Once we're confident about the API (props) for this component, we can add unit tests for it.